neqo-transport/src/recovery.rs

```diff
@@ -209,12 +227,152 @@ impl LossRecoverySpaces {
     }
 }
 
+#[derive(Debug)]
+struct CongestionControl {
+    congestion_window: u64, // = kInitialWindow
+    bytes_in_flight: u64,
+    congestion_recovery_start_time: Option<Instant>,
+    ssthresh: u64,
+}
+
+impl Default for CongestionControl {
+    fn default() -> Self {
+        CongestionControl {
+            congestion_window: INITIAL_WINDOW,
+            bytes_in_flight: 0,
+            congestion_recovery_start_time: None,
+            ssthresh: u64::max_value(),
+        }
+    }
+}
+
+impl Display for CongestionControl {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "CongCtrl")
+    }
+}
+
+impl CongestionControl {
+    #[cfg(test)]
+    pub fn cwnd(&self) -> u64 {
+        self.congestion_window
+    }
+
+    fn cwnd_avail(&self) -> u64 {
+        // BIF can be higher than cwnd due to PTO packets, which are sent even
+        // if avail is 0, but still count towards BIF.
+        self.congestion_window.saturating_sub(self.bytes_in_flight)
+    }
+
+    // OnPacketAckedCC
+    fn on_packet_acked(&mut self, pkt: &SentPacket) {
```

Comment:
There is a little bit of a non-uniform interface here: We ACK one packet, but mark loss for multiple packets.  Can you try to clean that up?

I would say that the multiple packet form is probably ideal as a baseline, but you might also have a single-packet function for each.

Reply:
Sure, although do note this will move us slightly further away from matching the -recovery pseudocode. (We already do not match it 100%)